### PR TITLE
Version 2 of ETKDG

### PR DESCRIPTION
**Pull request by sriniker**
This is to incorporate the updated version of the torsion patterns (Guba et al., J. Chem. Inf. Model. 56, 1-5, 2016) as version 2. 
The default remains version 1 (Schärfer et al., J. Med. Chem. 56, 2016-2028, 2013).

### CODE REVIEW

**Comment by greglandrum** [comment]
Thanks @sriniker. I hope it's ok with you that I'm planning to defer this to the next (2018_03_1) release?

**Comment by sriniker** [comment]
That’s fine with me.

File: Code/GraphMol/DistGeomHelpers/Embedder.h [hunk 1]
```diff
@@ -108,7 +96,8 @@ int EmbedMolecule(ROMol &mol, unsigned int maxIterations = 0, int seed = -1,
                   bool ignoreSmoothingFailures = false,
                   bool enforceChirality = true,
                   bool useExpTorsionAnglePrefs = false,
-                  bool useBasicKnowledge = false, bool verbose = false,
+                  bool useBasicKnowledge = false,
+				  int ETversion = 1, bool verbose = false,
```
Comment (greglandrum) [inline Code/GraphMol/DistGeomHelpers/Embedder.h]: Now that we have the versions that take the `EmbedParameters` object, I wonder if it's really worth continuing to add arguments to these functions. I'd argue that we should freeze the API on these and just support new options via `EmbedParameters`.

File: Code/GraphMol/DistGeomHelpers/Embedder.cpp [hunk 2]
```diff
@@ -675,14 +678,14 @@ int EmbedMolecule(ROMol &mol, unsigned int maxIterations, int seed,
                   const std::map<int, RDGeom::Point3D> *coordMap,
                   double optimizerForceTol, bool ignoreSmoothingFailures,
                   bool enforceChirality, bool useExpTorsionAnglePrefs,
-                  bool useBasicKnowledge, bool verbose, double basinThresh,
-                  bool onlyHeavyAtomsForRMS) {
+                  bool useBasicKnowledge, int ETversion, bool verbose,
```
[the remark this answers is not in the document]
Comment (sriniker) [reply Code/GraphMol/DistGeomHelpers/Embedder.cpp]: This is done now.

**Comment by sriniker** [comment]
The default in getParams is set and ETversion has been changed to unsigned int.

**Comment by sriniker** [comment]
The code has been changed such that version 2 of ETKDG is only via EmbedParams available. I have created a new short cut called ETKDGv2. 
A regression test has been added. For the specific molecule, version 1 and 2 of ETKDG should give the same result.

File: Code/GraphMol/DistGeomHelpers/testDgeomHelpers.cpp [hunk 3]
```diff
@@ -1727,6 +1727,27 @@ void testEmbedParameters() {
     delete mol;
   }
   {
+      std::string fname =
```
Comment (greglandrum) [inline Code/GraphMol/DistGeomHelpers/testDgeomHelpers.cpp]: This test claims to be using ETKDGv2, but it's using the reference data from ETKDG.
If this is an example where the two versions return exactly the same results, this is a bad test. If they should return different results then maybe you forgot to check in the reference file and update this test?

**Comment by greglandrum** [comment]
@sriniker : I'm going to do some refactoring here in order to simplify things and create a PR against your branch to make sure you're ok with it. 
I just made a comment about problems with one of your tests, could you please take a look and see what you think there?

**Comment by greglandrum** [comment]
Thanks for this @sriniker !

**Pull request merged**